src/consul.rs

```diff
@@ -490,5 +547,14 @@ async fn ensure_config_exists(
         index.replace(new_index);
     };
 
-    Ok(())
+    let res = res
+        .json::<Vec<KVResponse>>()
+        .await
+        .context("Failed to parse KV response")?;
+
+    ensure!(res.len() == 1, "Expected 1 KV response, got {}", res.len());
+
+    res[0]
+        .session
+        .ok_or_else(|| anyhow!("Key was not locked by any session"))
```

Comment:
Would it be nicer to
```rust
res.first().expect("Expected blah").session...
```
?
Then we can be safe within the same expression as opposed to also requiring the `ensure!()`.

Reply:
:+1: